occlusion.html

```diff
@@ -0,0 +1,708 @@
+        gl.shaderSource(fragmentShaderBBox, fsSourceBBox);
+        gl.compileShader(fragmentShaderBBox);
+
+        if (!gl.getShaderParameter(fragmentShaderBBox, gl.COMPILE_STATUS)) {
+            console.error(gl.getShaderInfoLog(fragmentShaderBBox));
+        }
+
+        var programBBox = gl.createProgram();
+        gl.attachShader(programBBox, vertexShaderBBox);
+        gl.attachShader(programBBox, fragmentShaderBBox);
+        gl.linkProgram(programBBox);
+
+        if (!gl.getProgramParameter(programBBox, gl.LINK_STATUS)) {
+            console.error(gl.getProgramInfoLog(programBBox));
+        }
+
+        var vsSourceAssistCam =  document.getElementById("vertex-assist-cam").text.trim();
+        var fsSourceAssistCam =  document.getElementById("fragment-assist-cam").text.trim();
+
+        var vertexShaderAssistCam = gl.createShader(gl.VERTEX_SHADER);
+        gl.shaderSource(vertexShaderAssistCam, vsSourceAssistCam);
+        gl.compileShader(vertexShaderAssistCam);
+
+        if (!gl.getShaderParameter(vertexShaderAssistCam, gl.COMPILE_STATUS)) {
+            console.error(gl.getShaderInfoLog(vertexShaderAssistCam));
+        }
+
+        var fragmentShaderAssistCam = gl.createShader(gl.FRAGMENT_SHADER);
+        gl.shaderSource(fragmentShaderAssistCam, fsSourceAssistCam);
+        gl.compileShader(fragmentShaderAssistCam);
+
+        if (!gl.getShaderParameter(fragmentShaderAssistCam, gl.COMPILE_STATUS)) {
+            console.error(gl.getShaderInfoLog(fragmentShaderAssistCam));
+        }
+
+        var programAssistCam = gl.createProgram();
+        gl.attachShader(programAssistCam, vertexShaderAssistCam);
+        gl.attachShader(programAssistCam, fragmentShaderAssistCam);
+        gl.linkProgram(programAssistCam);
+
+        if (!gl.getProgramParameter(programAssistCam, gl.LINK_STATUS)) {
+            console.error(gl.getProgramInfoLog(programAssistCam));
+        }
+
+        /////////////////////////
+        // GET UNIFORM LOCATIONS
+        /////////////////////////
+
+        var sceneUniformsLocation = gl.getUniformBlockIndex(program, "SceneUniforms");
+        gl.uniformBlockBinding(program, sceneUniformsLocation, 0);
+
+        var modelMatrixLocation = gl.getUniformLocation(program, "uModel");
+        var texLocation = gl.getUniformLocation(program, "tex");
+
+        var sceneUniformsLocationBBox = gl.getUniformBlockIndex(programBBox, "SceneUniforms");
+        gl.uniformBlockBinding(programBBox, sceneUniformsLocationBBox, 0);
+
+        var modelMatrixLocationBBox = gl.getUniformLocation(programBBox, "uModel");
+
+        var viewProjMatrixLocationAssistCam = gl.getUniformLocation(programAssistCam, "uViewProj");
+        var modelMatrixLocationAssistCam = gl.getUniformLocation(programAssistCam, "uModel");
+
+        /////////////////////
+        // SET UP GEOMETRY & scene
+        /////////////////////
+
+        var positionBuffer, uvBuffer, normalBuffer, indices;
+
+        // sphere high poly
```

Comment:
I'm doing this trying to mimic the real use case, where usually each node has a different geometry and context switch is happening, and try to enlarge the optimization.
Yeah, in terms of a sample, it would be neat to keep only one geometry. It's up to you.